config/airflow_local_settings.py

```diff
@@ -1,11 +1,47 @@
+from copy import deepcopy
+
+from kubernetes.client import models as k8s
+from kubernetes.client.models import V1Container, V1Pod
+
 STATE_COLORS = {
-    "queued": 'gray',
-    "running": 'lime',
-    "success": '#0000FF',
-    "failed": 'red',
-    "up_for_retry": 'gold',
-    "up_for_reschedule": 'turquoise',
-    "upstream_failed": 'orange',
-    "skipped": 'pink',
-    "scheduled": 'tan',
+    "queued": "gray",
+    "running": "lime",
+    "success": "#0000FF",
+    "failed": "red",
+    "up_for_retry": "gold",
+    "up_for_reschedule": "turquoise",
+    "upstream_failed": "orange",
+    "skipped": "pink",
+    "scheduled": "tan",
 }
+
+
+def pod_mutation_hook(pod: V1Pod):
+    """https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/V1Pod.md"""
+
+    # check that we're running a prio-processor job, and spin up a side-car
+    # container for proxying gcs buckets.
+    if pod.metadata.labels["job-kind"] == "prio-processor":
+        pod.spec.share_process_namespace = True
+        # there is only one container within the pod, so lets append a few more
+
+        # for proxying gcs resource consistently across platforms
+        minio_container = deepcopy(pod.spec.containers[0])
+        minio_container.image = "minio/minio:RELEASE.2021-06-17T00-10-46Z"
+        minio_container.args = ["gateway", "gcs", "$(PROJECT_ID)"]
+        minio_container.name = "minio"
+        pod.spec.containers.append(minio_container)
+
+        # Search for a new process named `minio-done` and kill the minio
+        # container above. This can be done using `exec -a minio-done sleep 10`
+        # which will will create a process available in the shared namespace for
+        # 10 seconds.
+        pkill_container = deepcopy(pod.spec.containers[0])
```

Comment:
you are using this copy, modify, append logic more than once, it may justify writing a helper method (or use the kubernetes libraries to construct an object from scratch. There may be some metadata or other extra things that have implications